source-firestore/main.go

```diff
@@ -0,0 +1,411 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	firestore "cloud.google.com/go/firestore"
+	firebase "firebase.google.com/go"
+	schemagen "github.com/estuary/connectors/go-schema-gen"
+	"github.com/estuary/connectors/schema_inference"
+	"github.com/estuary/flow/go/protocols/airbyte"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type State struct {
+	// Last time a full table scan was run, we use this to know if we need to do a full scan
+	// to reach eventual consistency
+	LastScan string
+}
+
+type config struct {
+	// Service account JSON key to use as Application Default Credentials
+	CredentialsJSON string `json:"googleCredentials" jsonschema:"title=Credentials,description=Google Cloud Service Account JSON credentials." jsonschema_extras:"secret=true,multiline=true"`
+
+	// How frequently should we scan all collections to ensure consistency
+	ScanInterval string `json:"scan_interval" jsonschema:"title=Scan Interval,description=How frequently should all collections be scanned to ensure consistency. See https://pkg.go.dev/time#ParseDuration for supported values.,default=12h"`
```

Comment:
Would it make sense to allow some sort of "never do a full scan" value here (the obvious options being a zero duration or the literal string `"never"`)?

Reply:
the literal string sounds good 👍🏽